src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -320,56 +331,65 @@ public void renameSheet(String name, String newName) {
     }
 
     public void cloneSheet(int index, String newName) {
-        cloneSheet(workbook.getSheetName(index), newName);
+        Sheet sheet = workbook.cloneSheet(index);
+        String originalName = workbook.getSheetName(index);
+        workbook.setSheetName(workbook.getSheetIndex(sheet), newName);
+        //Copy names (named ranges) that are scoped to the original sheet, adapting the formula and scope.
+        List<Name> originalNamedRanges = workbook.getAllNames().stream().filter(name -> name.getSheetIndex() == index).collect(Collectors.toList());
+        // we have to collect the original names *then* add the new names, otherwise we get concurrency issues creating names while iterating over them
+        originalNamedRanges.forEach(namedRange ->
+                createName(namedRange.getNameName(), newName, namedRange.getRefersToFormula().replace(originalName, newName), false));
     }
 
     public void cloneSheet(String name, String newName) {
-        Sheet sheet = workbook.cloneSheet(workbook.getSheetIndex(name));
-        workbook.setSheetName(workbook.getSheetIndex(sheet), newName);
+        cloneSheet(workbook.getSheetIndex(name), newName);
     }
-    
-    public void createName(String name, String formula, boolean overwrite) {
-        if(existsName(name)) {
+
+    public void createName(String name,  String worksheetScope, String formula, boolean overwrite) {
```

Comment:
If I'm not mistaken, the `formula` may include the worksheet name, in which case `worksheetScope` would be redundant?

Reply:
`formula` may target a specific sheet, but the name itself would be defined in `worksheetScope`. This is at least feasible manually in libreoffice:
![image](https://github.com/miraisolutions/xlconnect-java/assets/19765083/3d88c39c-afe4-43eb-be9f-737e7b02943d)